bionemo-recipes/recipes/llama_native_te_nvfsdp/dataloader.py

```diff
@@ -0,0 +1,276 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+
+"""
+Simple LLAMA3 Genomic Dataloader (no Lightning dependencies).
+
+Implements meeting requirements:
+1. First tiling, then tokenization  
+2. Fixed window positions + randomized access (EVO2 style)
+3. SQL backend for speed
+4. EVO2-style overlap ratio (200bp default)
+"""
+
+import sqlite3
+import random
+import numpy as np
+from typing import Dict, Optional, Tuple, Iterator
+from pathlib import Path
+import torch
+from torch.utils.data import Dataset, DataLoader, DistributedSampler
+from omegaconf import DictConfig
+
+from tokenizer import NucleotideASCIITokenizer
+
+
+class GenomicSequenceDataset(Dataset):
```

Comment:
This dataset class isn't quite right; high-level if we want to be consistent with the ESM-2 recipe (and generally follow the huggingface flow) we'd want a dataset that doesn't do tokenization, it just returns the full sequences. I'm not seeing a reason that couldn't essentially be a bunch of sharded parquet files with the raw sequences in them, like we do for esm-2:

https://huggingface.co/datasets/nvidia/esm2_uniref_pretraining_data

then you'd want various map operations to tokenize the dataset from there, and possibly a sampler to shuffle it.